docs/blog/2023/creating-custom-extension-for-tiptap-rte/index.md

```diff
@@ -0,0 +1,286 @@
+
+```<a href="mailto:info@ds.pl" ... >Contact Us</a>```
+
+From a user perspective, RTE should have a dedicated button where after clicking there is a possibility to input an email address, and the rest is happening under the hood. The button could be placed next to Link button. This is how RTE component looks like now:
+
+<p align="center" width="100%">
+    <img class="image" src="before-email-extension.png" alt="Extension for tiptap rich text editor in websight CMS">
+</p>
+
+Please keep in mind it is only a very basic method of email link obfuscation, chosen as one from many other ways of encoding link content. Surely it will not protect from all spambots or harvesters, but should be enough for our current needs.
+
+## Technical overview
+
+Due to component specifcity, the work can be split into two parts:
+
+- CMS part - where adding/editing/deleting and encrypting email addresses is happening within Rich Text Editor
+- page part - where RTE configuration and email decoding happens 
+
+Therefore, **the component must be used with the script provided within page**, otherwise, email links won’t be decoded.
+
+As the decoding email script will be the last step, let’s focus on the main issue here – extending the RTE editor. Such an editor component is made up of two parts: the plugin element and the UI element. The plugin part is responsible for the plugin behavior and it is directly related to the UI part, which is providing UI for the plugin itself. In Websight CMS, the tiptap library is responsible for plugins, and for UI - a separate module that is a part of Websight CMS. As for the plugin, we can either extend or create a new extension. But what about the UI element for this plugin? Any of these (button, button dropdown, list dropdown, link) don't match the requirements, because we need a dropdown with one input (link UI has input and select with `target` attribute selection). So we need a new UI as well.
+
+Our new plugin will affect another plugin that is already in the default version in RTE - Link. **We need to exclude email handling from the Link plugin**, as we will cover that in the email plugin. And that means we need also to modify the Link plugin by creating a custom link plugin.
+
+## Page part - configuration
+
+We can start by adding proper fields in the JSON file with richtext configuration. Under the `"configuration"` property we want to inherit all other components, so we can make use of the `"sling:resourceSuperType"` property and type our default path to richtext configuration `wcm/dialogs/components/richtext/configuration`. Afterward, we can add any plugin we want and in our case, `link` will be overwritten and `email` will be added as new. To not confuse and overcomplicate, let's keep simple titles and plugin/UI names. To keep both UI next to each other, let's use the `"sling:orderBefore"` property.
```

Comment:
Overriding a `.../application/backend/src/main/resources/apps/wcm/dialogs/components/richtext/configuration/.content.json` was a specific use of other sling mechanisms to force reusing new configuration instead of the default one. I think we can forget about it here and focus on preparing an extended configuration. So: 

- we can write that we want to prepare an extended configuration according to the documentation: https://www.websight.io/docs/developers/development/dialogs/richtext-editor/configuration/#extending-and-overriding-configuration
- we can use a "/apps/rte/extended/configuration/.content.json" (or something like that) as a path for an extended configuration
- mention that, we can use this new configuration in our components following the documentation: https://www.websight.io/docs/developers/development/dialogs/richtext-editor/configuration/#using-configuration